src/base.jl

```diff
@@ -49,14 +49,24 @@ Base.convert(::String, ss::ShortString) = String(ss)
 Base.display(s::ShortString) = display(String(s))
 Base.firstindex(::ShortString) = 1
 Base.isvalid(s::ShortString, i::Integer) = isvalid(String(s), i)
-Base.iterate(s::ShortString) = iterate(String(s))
-Base.iterate(s::ShortString, i::Integer) = iterate(String(s), i)
 Base.lastindex(s::ShortString) = sizeof(s)
 Base.ncodeunits(s::ShortString) = sizeof(s)
 Base.print(s::ShortString) = print(String(s))
 Base.show(io::IO, str::ShortString) = show(io, String(str))
-Base.sizeof(s::ShortString{T}) where T = Int(s.size_content & (size_mask(s) % UInt))
 
+@inline _get_word(s::ShortString{T}, i::Int) where {T} =
+           (s.size_content >> (8*(sizeof(T) - i - 3)))%UInt32
+
+@inline function Base.iterate(s::ShortString, i::Int=1)
+       0 < i <= ncodeunits(s) || return nothing
+       chr = _get_word(s, i)
+       chr < 0x8000_0000 ? (reinterpret(Char, chr & 0xFF00_0000), i + 1) :
+           chr < 0xe000_0000 ? (reinterpret(Char, chr & 0xFFFF_0000), i + 2) :
+           chr < 0xf000_0000 ? (reinterpret(Char, chr & 0xFFFF_FF00), i + 3) :
+           (reinterpret(Char, chr), i + 4)
```

Comment:
Purely a style comment but i find for multiline `?` statmeents `if`, `elseif`, `elseif` is much easier to read.